pkg/daemon/ceph/client/config.go

```diff
@@ -133,6 +140,48 @@ func generateConfigFile(context *clusterd.Context, clusterInfo *ClusterInfo, pat
 	return filePath, nil
 }
 
+func mergeDefaultConfigWithRookConfigOverride(context *clusterd.Context, clusterInfo *ClusterInfo, configFile *ini.File) error {
+	cm, err := context.Clientset.CoreV1().ConfigMaps(clusterInfo.Namespace).Get(k8sutil.ConfigOverrideName, metav1.GetOptions{})
```

Comment:
Instead of reading from a configmap, the preferred approach in the osd prepare jobs is that the configmap would be mounted as a volume. In most daemons it is mounted to /etc/ceph/ceph.conf, but it is not for the osd prepare job because we need to generate the config. Now we could mount it to a different path, then perform a merge with the generated config based on the file.

Reply:
It seems that this method of using the generated config is not only used by the osd prepare job (and could be in the future used by other).
It looks to me less practical to have to define the configmap mounted as a volume on all jobs using this generated config and to not forget of doing so for each new usage.
Or perhaps we only want this for the osd prepare job?

Reply:
The OSD prepare pod should be the only affected place. All of the other ceph daemons directly mount the ceph.conf from the configmap, and specify all other options on the command line in the pod spec.

Since the OSD prepare pod already has the RBAC to access the configmaps, agreed it would be simpler as you have it. I'm ok to keep it simple like this instead of going to the trouble of mounting the configmap. Then if anything besides the osd prepare job uses it in the future, it would also inherit for free without mounting it.

Reply:
FI this ceph.conf creation mechanism is at least also used by the removeOSD job and by the operator to create conf file in `/var/lib/rook/rook-ceph/rook-ceph.config` (for internal and external ceph cluster)

Reply:
Yes, those will work well with this change as well